src/Microsoft.Data.Sqlite.Core/SqliteValueBinder.cs

```diff
@@ -151,5 +202,123 @@ public virtual void Bind()
                 throw new InvalidOperationException(Resources.UnknownDataType(type));
             }
         }
+
+        internal static SqliteType GetSqliteType(object value)
+        {
+            if (value == null)
+            {
+                return SqliteType.Text;
+            }
+
+            var type = value.GetType().UnwrapNullableType().UnwrapEnumType();
```

Comment:
use switch on type and pattern matching of C# 7.0

Reply:
is switch is not possible then dictionary may be

Reply:
+1 for using a dictionary

Reply:
Changed to using a dictionary in the newest version.